components/no_results_indicator/no_results_indicator.tsx

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage, MessageDescriptor} from 'react-intl';
+
+import {t} from 'utils/i18n';
+
+import searchImage from 'images/search.png';
+
+import {NoResultsVariant} from './types';
+
+interface Props {
+    variant: NoResultsVariant;
+    value?: {[key: string]: string};
+}
+
+const iconMap: {[key in NoResultsVariant]: string } = {
+    [NoResultsVariant.Search]: searchImage,
+
+    // this will be changed soon in the PRs following the other no results indicators
+    [NoResultsVariant.Mentions]: '',
+    [NoResultsVariant.FlaggedPosts]: '',
+    [NoResultsVariant.PinnedPosts]: '',
+};
+
+const titleMap: {[key in NoResultsVariant]: MessageDescriptor } = {
+    [NoResultsVariant.Search]: {
+        id: t('no_results.channels.title')
```

Comment:
Feels odd to have a channel specific format string inside a generic component, no? I imagine this component being used in search scenarios as well, like text search in RHS, no?

If that's the case I wonder if we can pass this through the props? would help with the customizability, thoughts? 🤔

Reply:
I can change the string to make it specific to the quick switch modal, I thought I would use the same text for the Channel Search in the RHS, but later saw that the text was different :D So I can add a new variant for the RHS. 

I created the Enum in order to make it customisable and simple to use enough by passing a Variant to props only. Otherwise it looked quite weird to pass all the different formatted message values through props (ex. titleId, subtitleId, defaultMessageTitle, defaultMessagesubtitle etc.) and this is handled quite well by the messageDescriptor.

Reply:
I see. Fair enough. Only issue is as soon as we have more customized types this list of variants will grow :P, which isn't a bad thing.

In which case then do you mind renaming the `Prop.value` to something that's close to FromattedMessage, otherwise who ever wants to use this component will first have to look at how `value` prop is being used and then use it. Could name it something like `formattedMessageValues`, what do you think?